tests/frontier/scenarios/test_scenarios.py

```diff
@@ -0,0 +1,261 @@
+
+    operation = replace_special_calls_in_operation(pre, operation, external_address)
+
+    combinations_input: ScenarioGeneratorInput = ScenarioGeneratorInput(
+        fork=fork,
+        pre=pre,
+        operation_code=operation,
+        external_address=external_address,
+        external_balance=external_balance,
+    )
+
+    call_combinations = ScenariosCallCombinations(combinations_input).generate()
+    for combination in call_combinations:
+        if not debug.scenario_name or combination.name == debug.scenario_name:
+            scenarios_list.append(combination)
+
+    call_combinations = scenarios_create_combinations(combinations_input)
+    for combination in call_combinations:
+        if not debug.scenario_name or combination.name == debug.scenario_name:
+            scenarios_list.append(combination)
+
+    revert_combinations = scenarios_revert_combinations(combinations_input)
+    for combination in revert_combinations:
+        if not debug.scenario_name or combination.name == debug.scenario_name:
+            scenarios_list.append(combination)
+
+    """
+        // 21. 0x00FD  Run the code, call a contract that reverts, then run again
+        // 22. 0x00FE  Run the code, call a contract that goes out of gas, then run again
+        // 23. 0x00FF  Run the code, call a contract that self-destructs, then run again
+        // 34. 0x60BACCFA57 Call recurse to the limit
+    """
+
+    return scenarios_list
+
+
+@pytest.mark.valid_from("Frontier")
+@pytest.mark.parametrize(
+    # select program to debug (program, "scenario_name")
+    # program=None select all programs
+    # scenario_name="" select all scenarios
+    "debug",
+    [ScenarioDebug(None, scenario_name="")],
+    ids=["debug"],
+)
+@pytest.mark.parametrize(
+    "operation, expected_result",
+    [
+        # invalid opcodes
+        program_invalid,
+        # static violation programs
+        program_sstore_sload,
+        program_tstore_tload,
+        program_logs,
+        # 00 - 20
+        program_all_frontier_opcodes,
+        # 30
+        program_address,
+        program_balance,
+        program_origin,
+        program_caller,
+        program_callvalue,
+        program_calldataload,
+        program_calldatasize,
+        program_calldatacopy,
+        program_codecopy_codesize,
+        program_gasprice,
+        program_ext_codecopy_codesize,
+        program_returndatasize,
+        program_returndatacopy,
+        program_extcodehash,
+        # 40
+        program_blockhash,  # fails in state mode in py-t8n
+        program_coinbase,
+        program_timestamp,
+        program_number,
+        program_difficulty_randao,
+        program_gaslimit,
+        program_chainid,
+        program_selfbalance,
+        program_basefee,
+        program_blobhash,
+        program_blobbasefee,
+        program_tload,
+        program_mcopy,
+        program_push0,
+        program_suicide,
+    ],
+)
+def test_scenarios(
```

Comment:
```suggestion
@pytest.mark.execute(pytest.mark.skip(reason="gas usage"))
def test_scenarios(
```
This test has to be skipped in execute mode because it uses 500 million gas.

Reply:
should not be that much. I need to correct

Reply:
can make a test per scenario. right now all scenarios combined in one test for each program. 
that would be a lot of test files though